Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -694,6 +694,10 @@ void ComplexTextController::adjustGlyphsAndAdvances()
                 // Like simple text path in WidthIterator::applyCSSVisibilityRules,
                 // make tabCharacter glyph invisible after advancing.
                 glyph = deletedGlyph;
+            } else if (character == zeroWidthNonJoiner) {
+                // zeroWidthNonJoiner is rendered as deletedGlyph for compatibility with other engines: https://bugs.webkit.org/show_bug.cgi?id=285959
+                advance.setWidth(0);
+                glyph = deletedGlyph;
```

Comment:
Can we write a test case for this?

Reply:
Yes. I intend to write a test for it, but I need to create a special Ahem font for it where that character is visible.

Reply:
Should this catch only ZWNJ, or should this check be for all default-ignorable characters?